lib/integrations/awsoidc/deployservice.go

```diff
@@ -445,16 +445,18 @@ func DeployService(ctx context.Context, clt DeployServiceClient, req DeployServi
 		return nil, trace.Wrap(err)
 	}
 
-	serviceDashboardURL := fmt.Sprintf("https://%s.console.aws.amazon.com/ecs/v2/clusters/%s/services/%s", req.Region, aws.ToString(req.ClusterName), aws.ToString(req.ServiceName))
-
 	return &DeployServiceResponse{
 		ClusterARN:          aws.ToString(cluster.ClusterArn),
 		ServiceARN:          aws.ToString(service.ServiceArn),
 		TaskDefinitionARN:   taskDefinitionARN,
-		ServiceDashboardURL: serviceDashboardURL,
+		ServiceDashboardURL: serviceDashboardURL(req.Region, aws.ToString(req.ClusterName), aws.ToString(req.ServiceName)),
 	}, nil
 }
 
+func serviceDashboardURL(region, clusterName, serviceName string) string {
+	return fmt.Sprintf("https://%s.console.aws.amazon.com/ecs/v2/clusters/%s/services/%s", region, clusterName, serviceName)
```

Comment:
can we make 100% sure that region, clusterName, and serviceName are validated before building a URL like this to avoid SSRF? Maybe they already are but it's not extremely clear to me where these are all coming from

Reply:
This happens in two places:
- after deploying a service
- when creating the list of already deployed services

So, it's always after those values were used to call the AWS API.
I'm pretty confident that invalid values do not reach this method given its current usage.

Anyway, I'll add checks ensuring that they are valid so that future usage of this method is also not prone to SSRF.